src/Microsoft.PowerShell.ConsoleHost/host/msh/ManagedEntrance.cs

```diff
@@ -30,19 +32,39 @@ public sealed class UnmanagedPSEntry
         /// <param name="argc">
         /// Length of the passed in argument array.
         /// </param>
+        [Obsolete("Callers should now use UnmanagedPSEntry.Start(string[])", error: true)]
         public static int Start(string consoleFilePath, [MarshalAs(UnmanagedType.LPArray, ArraySubType = UnmanagedType.LPWStr, SizeParamIndex = 2)]string[] args, int argc)
         {
+            return Start(args);
+        }
+
+        /// <summary>
+        /// Starts managed MSH.
+        /// </summary>
+        /// <param name="args">
+        /// Command line arguments to the managed MSH
+        /// </param>
+        public static int Start(string[] args)
```

Comment:
When it's called from native code, I'm afraid this signature won't work, at least on Windows. You will need the `[MarshalAs(...)]` attribute for this parameter. For backward compatible reason, I suggest to keep using the old method signature.

Or we can have a new one that removes `consoleFilePath`, but `SizeParamIndex` needs to be changed to 1.

Reply:
Added MarshalAs().